hard/tsmc_28/bsg_mem/bsg_mem_3r1w_sync_macros.vh

```diff
@@ -0,0 +1,183 @@
+                                                                \
+             /* According to TSMC, other settings are for debug only */ \
+             ,.WTSEL    ( 2'b01         )                       \
+             ,.RTSEL    ( 2'b01         )                       \
+             ,.VG       ( 1'b1          )                       \
+             ,.VS       ( 1'b1          )                       \
+            );                                                  \
+          tsmc28_2rw_d``words``_w``bits``_``tag``_2sram mem2    \
+            (                                                   \
+              .AA       ( w_addr_i      )                       \
+             ,.DA       ( w_data_i      )                       \
+             ,.WEBA     ( ~w_v_i        )                       \
+             ,.CEBA     ( ~w_v_i        )                       \
+             ,.CLKA     ( clk_i         )                       \
+             ,.QA       (               )                       \
+                                                                \
+             ,.AB       ( r2_addr_i     )                       \
+             ,.DB       ( '1            )                       \
+             ,.WEBB     ( '1            )                       \
+             ,.CEBB     ( ~r2_v_i       )                       \
+             ,.CLKB     ( clk_i         )                       \
+             ,.QB       ( r2_data_o     )                       \
+                                                                \
+             /* According to TSMC, other settings are for debug only */ \
+             ,.WTSEL    ( 2'b01         )                       \
+             ,.RTSEL    ( 2'b01         )                       \
+             ,.VG       ( 1'b1          )                       \
+             ,.VS       ( 1'b1          )                       \
+            );                                                  \
+  end
+
+`define bsg_mem_3r1w_sync_2hdsram_macro(words,bits,tag) \
+if (harden_p && els_p == words && width_p == bits)              \
+  begin: macro                                                  \
+          tsmc28_2rw_d``words``_w``bits``_``tag``_2hdsram mem0  \
+            (                                                   \
+              .AA       ( w_addr_i      )                       \
+             ,.DA       ( w_data_i      )                       \
+             ,.WEBA     ( ~w_v_i        )                       \
+             ,.CEBA     ( ~w_v_i        )                       \
+             ,.CLK      ( clk_i         )                       \
+             ,.QA       (               )                       \
+                                                                \
+             ,.AB       ( r0_addr_i     )                       \
+             ,.DB       ( '1            )                       \
+             ,.WEBB     ( '1            )                       \
```

Comment:
how many bits is this supposed to be? maybe be more explicit about the width?

Reply:
made explicit!